sound/soc/sof/pm.c

```diff
@@ -238,6 +242,9 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 
 suspend:
 
+	/* release trace */
+	snd_sof_release_trace(sdev, target_state == SOF_DSP_PM_D0 ? true : false);
```

Comment:
After giving this a thought I think we should aim a more generic approach by passing the `target_state` directly and let the dtrace code to decipher from there.
But the SOF_DSP_PM_D0 is not really the state we are entering (from SOF_DSP_PM_D0), but it is SOF_DSP_PM_D0:SOF_HDA_DSP_PM_D0I3, so is is more like SOF_DSP_PM_D0_LP.

Basically we will tell the dtrace that we are suspending to the given level:
SOF_DSP_PM_D3 - DSP is going to be off
SOF_DSP_PM_D0 - DSP remains on

Reply:
> After giving this a thought I think we should aim a more generic approach by passing the `target_state` directly and let the dtrace code to decipher from there.
> But the SOF_DSP_PM_D0 is not really the state we are entering (from SOF_DSP_PM_D0), but it is SOF_DSP_PM_D0:SOF_HDA_DSP_PM_D0I3, so is is more like SOF_DSP_PM_D0_LP.
> 
> Basically we will tell the dtrace that we are suspending to the given level:
> SOF_DSP_PM_D3 - DSP is going to be off
> SOF_DSP_PM_D0 - DSP remains on

@ujfalusi This should be another topic beyond this PR, which may impact the dtrace behavior and policy.

Reply:
@libinyang, the policy is:
if the DSP is going off then free up all resources and on resume set up everything from scratch
if the DSP is not going to be off then stop the host DMA and on resume only restart the host DMA

Doing the `target_state == SOF_DSP_PM_D0 ? true : false` in `pm.c` **or** in the `trace.c` does not change the policy and behavior especially that _this_ PR is changing it in the first place.

Reply:
@ujfalusi I mean the policy whether we should keep ```trace.c``` as simple as possible and won't take the responsibility to decide trigger stop or trigger start.

Reply:
The trace is moving out as a separate driver and it will have to make it's own decisions anyways.

Reply:
and it is not trigger start or stop question. It is suspend and resume question. Are suspending to a level where the DSP is off or are we keeping it on.

Reply:
> The trace is moving out as a separate driver and it will have to make it's own decisions anyways.

After reading the email you sent, it seems I understand your ideas now. Let's talk more in the weekly meeting.